client/store/products/reducer.js

```diff
@@ -0,0 +1,39 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+/**
+ * Internal dependencies
+ */
+import { ERROR } from 'store/constants';
+const DEFAULT_STATE = {
+	queries: {},
+};
+export default function productsReducer( state = DEFAULT_STATE, action ) {
+	if ( 'SET_PRODUCTS' === action.type ) {
+		const prevQueries = get( state, 'queries', [] );
+		const query = JSON.stringify( action.query, Object.keys( action.query ).sort() );
```

Comment:
Just a note that in #299, I'm adding a helper function for this: https://github.com/woocommerce/wc-admin/pull/299/files#diff-508644db164d1468748f412189a82e78. If you want to add it here first, that works too.